File: app/src/at-vinhnguyen/java/asiantech/internship/summer/javacore/exercise16/TheNumberOfOccurrencesOfTheElement.java
package asiantech.internship.summer.javacore.exercise16;

import android.util.Log;
import java.util.Map;
import java.util.TreeMap;

import static asiantech.internship.summer.javacore.InputUtils.inputIntNumber;

public class TheNumberOfOccurrencesOfTheElement {

    private static final String TAG = TheNumberOfOccurrencesOfTheElement.class.getSimpleName();

    public static void main(String[] args) {
        Log.d(TAG, "Nhập số phần tử của mảng: ");
        int n = inputIntNumber(TAG);
        int[] arr = new int[n];
        Log.d(TAG, "Nhập các phần tử của mảng: \n");
        for (int i = 0; i < n; i++) {
            Log.d(TAG, "a[" + i + "] = ");
            arr[i] = inputIntNumber(TAG);
        }
        Map<Integer, Integer> map = new TreeMap<>();
        for (int i = 0; i < n; i++) {
            addElement(map, arr[i]);
        }
        for (Integer key : map.keySet()) {
            Log.d(TAG, key + " xuất hiện " + map.get(key) + " lần.\n");
        }
    }

    private static void addElement(Map<Integer, Integer> map, int element) {
        if (map.containsKey(element)) {
            int count = map.get(element) + 1;
            map.put(element, count);
        } else {
            map.put(element, 1);
        }
    }
}

File: app/src/at-vinhnguyen/java/asiantech/internship/summer/kotlincore/exercise16/TheNumberOfOccurrencesOfTheElement.kt
package asiantech.internship.summer.kotlincore.exercise16

import inputIntNumber
import java.util.*

object TheNumberOfOccurrencesOfTheElement {

    @JvmStatic
    fun main(args: Array<String>) {
        print("Nhập số phần tử của mảng: ")
        val n = inputIntNumber()
        val arr = IntArray(n)
        print("Nhập các phần tử của mảng: \n")
        for (i in 0 until n) {
            System.out.printf("a[%d] = ", i)
            arr[i] = inputIntNumber()
        }
        val map = TreeMap<Int, Int>()
        for (i in 0 until n) {
          addElement(
              map, arr[i])
        }
        for (key in map.keys) {
            System.out.printf("%d xuất hiện %d lần.\n", key, map[key])
        }
    }

    private fun addElement(map: MutableMap<Int, Int>, element: Int) {
        if (map.containsKey(element)) {
            var count = map[element]!! + 1
            map[element] = count
        } else {
            map[element] = 1
        }
    }
}
